systemtest/src/main/java/io/strimzi/systemtest/cli/KafkaCmdClient.java

```diff
@@ -25,9 +30,16 @@ public KafkaCmdClient() { }
 
     public static String createTopicUsingPodCli(String clusterName, int zkPodId, String topic, int replicationFactor, int partitions) {
         String podName = KafkaResources.zookeeperPodName(clusterName, zkPodId);
-        return cmdKubeClient().execInPod(podName, "/bin/bash", "-c",
+        String response = cmdKubeClient().execInPod(podName, "/bin/bash", "-c",
             "bin/kafka-topics.sh --zookeeper localhost:" + PORT + " --create " + " --topic " + topic +
                 " --replication-factor " + replicationFactor + " --partitions " + partitions).out();
+
+        KafkaTopicUtils.waitForKafkaTopicCreation(topic);
+
+        KafkaTopic kafkaTopic = KafkaTopicResource.kafkaTopicClient().inNamespace(kubeClient().getNamespace()).withName(topic).get();
+        ResourceManager.getPointerResources().push(() -> ResourceManager.deleteLater(KafkaTopicResource.kafkaTopicClient(), kafkaTopic));
```

Comment:
I don't have better solution in mind atm, but in case if creation will fail, the deletion will not be executed. Can we delete topic based on name and maybe move this code before wait of line 37?

Reply:
Sure we can

Reply:
Just thinking about this -> if the creation will fail, so the topic will not be created, we don't need to delete it after test, as it will not be present, right? But it's true that this will prevent from race conditions.